src/boundingmesh/Mesh.cpp

```diff
@@ -189,367 +189,44 @@ void swap(Mesh& first, Mesh& second) {
-                      const SoPrimitiveVertex* v1, const SoPrimitiveVertex* v2,
-                      const SoPrimitiveVertex* v3) {
-  CallbackData* data = static_cast<CallbackData*>(userData);
-
-  SbMatrix* matrix = data->matrix;
-  VertexPositionSet* vertices = data->vertices;
-  Mesh* mesh = data->mesh;
-
-  SbVec3f points[3];
-
-  matrix->multVecMatrix(v1->getPoint(), points[0]);
-  matrix->multVecMatrix(v2->getPoint(), points[1]);
-  matrix->multVecMatrix(v3->getPoint(), points[2]);
-
-  Index indices[3];
-  for (unsigned int i = 0; i < 3; ++i) {
-    Vector3 position;
-    position << points[i][0], points[i][1], points[i][2];
-    indices[i] = vertices->addVertex(position);
-  }
-  if (indices[0] == indices[1] || indices[0] == indices[2] ||
-      indices[1] == indices[2]) {
-#ifndef NDEBUG
-    std::cout << "Skipping degenerate triangle " << indices[0] << ", "
-              << indices[1] << ", " << indices[2] << std::endl;
-#endif
-  } else
-    mesh->addTriangle(indices[0], indices[1], indices[2]);
-}
-#endif
-
-void Mesh::loadWrl(const std::string& filename, int faceset, bool debugOutput) {
-#ifndef COIN_AVAILABLE
-  std::cout << "Coin required, can't load .wrl." << std::endl;
-#else
-  SoDB::init();
-
-  SoGroup* root;
-  SoInput input;
-  if (!input.openFile(filename.c_str(), true)) {
-    std::cout << "Error: File not found." << std::endl;
-    exit(-1);
-  } else {
-    root = SoDB::readAll(&input);
-  }
-
-  input.closeFile();
-  if (NULL == root) {
-    std::cerr << "Error: Could not read file." << std::endl;
-    exit(-1);
-  }
-
-  root->ref();
-
-  SbViewportRegion viewportRegion;
-  SoSearchAction searchAction;
-  searchAction.setInterest(SoSearchAction::ALL);
-  searchAction.setType(SoShape::getClassTypeId());
-  searchAction.apply(root);
-
-  CallbackData cb_data;
-  cb_data.mesh = this;
-
-  if (debugOutput) {
-    std::cout << "Note: VRML contains " << searchAction.getPaths().getLength()
-              << " facesets." << std::endl;
-  }
-  for (int i = 0; i < searchAction.getPaths().getLength(); ++i) {
-    if (faceset != -1 && i != faceset) continue;
-
-    SoGetMatrixAction* getMatrixAction = new SoGetMatrixAction(viewportRegion);
-    getMatrixAction->apply(searchAction.getPaths()[i]);
-    SbMatrix matrix = getMatrixAction->getMatrix();
-
-    VertexPositionSet vertices(this);
-    cb_data.vertices = &vertices;
-    cb_data.matrix = &matrix;
-
-    SoCallbackAction callbackAction;
-    callbackAction.addTriangleCallback(SoShape::getClassTypeId(),
-                                       triangleCallback, &cb_data);
-    callbackAction.apply(searchAction.getPaths()[i]);
-  }
-#endif
-}
 
-void Mesh::loadStl(const std::string& filename) {
-  std::ifstream file(filename.c_str());
-  std::string start;
-  file >> start;
-  std::cout << "start: " << start << std::endl;
-  bool binary = true;
-  if (start == "solid") binary = false;
-  file.close();
+  int currentIndexFace = 0;
```

Comment:
use our format style: read https://github.com/gaschler/bounding-mesh#contribute closely, install and execute clang-format as documented